### DISCUSSION TITLE
Is it possible to compress/minify js files in the public folder?

### DISCUSSION THREAD

**Discussion by churichard** [vercel/next.js#12614] [Help] [answered]
I'm looking to serve a static js file that users can embed on their website.

I currently have a .js file in my public folder and it works fine. However, the js file is not compressed/minified so the entire source code is inspectable and quite large.

How can I compress/minify js files in the public folder?

**Comment by ArtyomResh** [comment]
@churichard Hey! Did you found the answer?

**Comment by jamesmosier** [comment, marked as answer]
There is no built-in way in Next to minify any code within the `public` folder. You would need to do this yourself using a custom script before you build. 

The `public` folder was really just meant for other static resources (i.e. images) so built-in minification of JavaScript files was not apart of that spec.

**Reply by ArtyomResh** [reply]
Thank you! This is exactly what i did in the end :)

**Reply by danielLee0721** [reply]
@ArtyomResh How did you go about accomplishing this?

Any help would be appreciated. Thanks :)

**Reply by Wattyyy** [reply]
@danielLee0721 
Maybe you have already accomplished this, but in my case, I minified custom scripts with [esbuild](https://github.com/evanw/esbuild) as follows.

```
// package.json
"dev": "esbuild public/js/*.js --outdir=public/js/ --allow-overwrite --minify; next dev",
"build": "esbuild public/js/*.js --outdir=public/js/ --allow-overwrite --minify; next build",
```

**Reply by Mecanik** [reply]
Why `_next-assets`?

**Reply by Wattyyy** [reply]
@Mecanik 
I copied the `package.json` command from my old project, but now that I think about it, we don't need `_next-assets`. So, I edited the command.
